op/main.py

```diff
@@ -44,46 +45,50 @@ def _handle_event_link(charm_dir, bound_event):
         # (otherwise, we would never have been called).
         return
 
-    event_hook_path = charm_dir / 'hooks' / bound_event.event_kind.replace('_', '-')
+    if issubclass(bound_event.event_type, op.charm.HookEvent):
+        event_dir = charm_dir / 'hooks'
+    else:
+        raise RuntimeError(f'unable to create a symlink: unsupported event type {bound_event.event_type}')
+    # TODO: Handle function/action events here.
+
+    if not event_dir.exists():
+        raise RuntimeError(f'unable to create event symlink: {event_dir} directory does not exist')
```

Comment:
s/unable to/cannot/

Why isn't it trying to create the directory, though?  If it's going as far as creating the symlink itself, don't see a good reason not to put the directory in place as well.

Reply:
We expect at least an `install` or `start` symlink to be there to get a location of the charm code file from there (especially since, as you pointed out in https://github.com/canonical/operator/pull/63/files#r352619084, we shouldn't hardcode the relative symlink value in the `main.py` module itself).

Considering that, creating a directory like this only makes some sense for `functions`/`actions` which means we will need to special-case that.

Reply:
I would leave this as-is until we get to adding `functions`.